src/kloewe/dot/src/dot.h

```diff
@@ -0,0 +1,92 @@
+/*----------------------------------------------------------------------------
+  File    : dot.h
+  Contents: dot product (cpu dispatcher)
+  Author  : Kristian Loewe
+----------------------------------------------------------------------------*/
+#ifndef DOT_H
+#define DOT_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*----------------------------------------------------------------------------
+  Enum to encode the sets of dot implementations
+----------------------------------------------------------------------------*/
+enum flags {
+    DOT_NAIVE  = 1,
+    DOT_SSE2   = 2,
+    DOT_AVX    = 3,
+    #ifndef DOT_NOFMA
+    DOT_AVXFMA = 4,
+    #endif
+    DOT_AUTO   = 3
```

Comment:
That is interesting.  Personally, I would reorder the values according to how fast they benchmark, and make AUTO a much larger value, but I don't feel that strongly about it.

Either way, please add a comment or two (on the enum and in dot_set_impl) explaining the situation.